### PR TITLE
Feature/improve fedimg details

### PR DESCRIPTION
**Pull request by oddshocks**
With this PR, a little extra info will be printed out for Fedimg jobs, when applicable. Users will be able to see AMI ID, virtualization type, and volume type in the Fedmsg subtitle.

Also updated the tests for Fedimg stuff and added a new test. Tests run without error, but I could have made a mistake somewhere, or maybe there's something I could have done better? :)

### CODE REVIEW

File: fedmsg_meta_fedora_infrastructure/fedimg.py [hunk 1]
```diff
@@ -35,31 +34,39 @@ def subtitle(self, msg, **config):
             if msg['msg']['status'] == "started":
                 name = msg['msg']['image_name']
                 dest = msg['msg']['destination']
-                tmpl = self._('Image {image_name} started uploading to {dest}')
+                tmpl = self._('{image_name} started uploading to to {dest}')
```
Comment (ralphbean) [inline fedmsg_meta_fedora_infrastructure/fedimg.py]: "uploading **to** to {dest}"?

File: fedmsg_meta_fedora_infrastructure/fedimg.py [hunk 2]
```diff
@@ -35,31 +34,39 @@ def subtitle(self, msg, **config):
...
                 return tmpl.format(image_name=name, dest=dest)
             elif msg['msg']['status'] == "completed":
                 name = msg['msg']['image_name']
                 dest = msg['msg']['destination']
-                tmpl = self._('Image {image_name} finished uploading to to {dest}')
-                return tmpl.format(image_name=name, dest=dest)
+                extra = ', '.join(msg['msg']['extra'].keys())
+                tmpl = self._('{image_name} finished uploading to to {dest} '
```
Comment (ralphbean) [inline fedmsg_meta_fedora_infrastructure/fedimg.py]: Same here.

File: fedmsg_meta_fedora_infrastructure/tests/fedimg.py [hunk 3]
```diff
@@ -57,15 +56,55 @@ class TestImageUpload(Base):
...
+    expected_title = "fedimg.image.upload"
+    image_name = "fedora-cloud-base-rawhide-20140604.x86_64"
+    dest = "EC2-eu-west-1"
+    ami_id = 'ami-1234fda'
+    virt_type = 'HVM'
+    vol_type = 'gp2'
+    expected_subti = "{0} finished uploading to {1} ({2}, {3}, {4})".format(
+            image_name, dest, ami_id, virt_type, vol_type)
+    expected_link = None
+    expected_icon = None
```
Comment (ralphbean) [inline fedmsg_meta_fedora_infrastructure/tests/fedimg.py]: Seems like the icon should not be ``None`` here.

File: fedmsg_meta_fedora_infrastructure/tests/fedimg.py [hunk 4]
```diff
@@ -57,15 +56,55 @@ class TestImageUpload(Base):
...
+    expected_secondary_icon = None
+    expected_packages = set([])
+    expected_usernames = set([])
+    expected_objects = set([])
```
Comment (ralphbean) [inline fedmsg_meta_fedora_infrastructure/tests/fedimg.py]: No ``objects``?

File: fedmsg_meta_fedora_infrastructure/tests/fedimg.py [hunk 5]
```diff
@@ -26,14 +26,13 @@
 from .common import add_doc
 
 
-class TestImageUpload(Base):
-    """ These messages are awarded when an image starts has started,
-    completes, or fails. """
+class TestImageUploadStart(Base):
+    """ These messages are awarded when an image upload has started. """
```
Comment (ralphbean) [inline fedmsg_meta_fedora_infrastructure/tests/fedimg.py]: While we're here, could you add some more information to these docstrings about what fedimg does?  That'll make the fedmsg.com docs more readable.

File: fedmsg_meta_fedora_infrastructure/tests/fedimg.py (hunk 4 above)
Comment (oddshocks) [inline fedmsg_meta_fedora_infrastructure/tests/fedimg.py]: Sorry -- what sort of thing should be here?

File: fedmsg_meta_fedora_infrastructure/fedimg.py [hunk 6]
```diff
@@ -35,31 +34,31 @@ def subtitle(self, msg, **config):
             if msg['msg']['status'] == "started":
                 name = msg['msg']['image_name']
                 dest = msg['msg']['destination']
-                tmpl = self._('Image {image_name} started uploading to {dest}')
```
Comment (ralphbean) [inline fedmsg_meta_fedora_infrastructure/fedimg.py]: It seems like this will crash if you remove ``tmpl`` here.  Is this a typo?

Comment (oddshocks) [inline fedmsg_meta_fedora_infrastructure/fedimg.py]: Yeah, this was a mistake. It's fixed in the latest commit, where all the fedimg-related tests pass.

**Comment by oddshocks** [comment]
Ok, @ralphbean. Things should all be in order now, unless there's something else you'd like to see added to the docstrings or elsewhere. :)

**Comment by ralphbean** [comment]
:+1: to merge this @oddshocks.  Thanks!

**Pull request merged**